### PR TITLE
Add date validation using minValue and maxValue extensions

### PR DESCRIPTION
**Pull request by vishnuravi**
# Add date validation using minValue and maxValue extensions

## :recycle: Current situation & Problem
Currently ResearchKitOnFHIR does not support date validation, although this is supported by [Phoenix](https://github.com/StanfordBDHG/Phoenix) and also implemented in the [Android FHIR SDK](https://github.com/google/android-fhir) via the `http://hl7.org/fhir/StructureDefinition/minValue` and `http://hl7.org/fhir/StructureDefinition/maxValue` extensions.

## :gear: Release Notes 
Adds support for date validation via the `valueDate` property in the `http://hl7.org/fhir/StructureDefinition/minValue` and `http://hl7.org/fhir/StructureDefinition/maxValue` extensions.

## :books: Documentation
Please refer to the official HL7 FHIR extension specification for the minValue and maxValue extensions, which this PR follows to implement date validation:
- MinValue: http://hl7.org/fhir/StructureDefinition/minValue
- MaxValue: http://hl7.org/fhir/StructureDefinition/maxValue

## :white_check_mark: Testing
Unit tests and UI tests have been added.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vishnuravi** [comment]
The UI test seems to be failing because of a time zone issue. It appears that RK is interpreting the minimum date provided by the extension as being in UTC time and converting to the user's local time zone when applying the validation in the picker.

**Comment by vishnuravi** [comment]
The UI tests appear to be failing due to a time zone mismatch.

File: Sources/ResearchKitOnFHIR/FHIRToResearchKit/QuestionnaireItem+ResearchKit.swift [hunk 1]
```diff
@@ -166,9 +166,15 @@ extension QuestionnaireItem {
...
             return ORKTextChoiceAnswerFormat(style: choiceAnswerStyle, textChoices: answerOptions)
         case .date:
-            return ORKDateAnswerFormat(style: ORKDateAnswerStyle.date)
+            return ORKDateAnswerFormat(
+                style: .date,
+                defaultDate: nil,
+                minimumDate: minDateValue,
+                maximumDate: maxDateValue,
+                calendar: nil
+            )
```
Comment (Supereg) [inline Sources/ResearchKitOnFHIR/FHIRToResearchKit/QuestionnaireItem+ResearchKit.swift:175]: Thanks for tackling this 🚀

**Review by Supereg — APPROVED** [review]
Looks great 👍 

As discussed, I think we should think about a solution to have relative min/max days to specify constraints like "Date not later than today" which is super helpful for e.g. birthday pickers.

The extensions Phoenix currently uses for that are relatively hard to parse for machines so it might just make sense to define our own extension for that with a simpler encoding and decoding method.

**Comment by vishnuravi** [comment]
> Looks great 👍
> 
> As discussed, I think we should think about a solution to have relative min/max days to specify constraints like "Date not later than today" which is super helpful for e.g. birthday pickers.
> 
> The extensions Phoenix currently uses for that are relatively hard to parse for machines so it might just make sense to define our own extension for that with a simpler encoding and decoding method.

Agreed! Let's figure out the details and get it together in the next PR!
